src/encoded/static/components/facets/biochemical_inputs.js

```diff
@@ -0,0 +1,81 @@
+import PropTypes from 'prop-types';
+import FacetRegistry from './registry';
+import QueryString from '../../libs/query_string';
+
+
+/**
+ * Biochemical input facet works just like a regular facet but negating is not allowed, links look more like buttons, and there is an "All" button to select all possible terms
+ */
+const BiochemicalInputsFacet = ({ facet, relevantFilters, queryString }, reactContext) => {
+    const initializeBiochemicalInput = relevantFilters.map((filter) => filter.term);
+    const facetObj = {};
+    facet.terms.forEach((term) => {
+        const query = new QueryString(queryString);
+        facetObj[term.key] = {};
+        if (initializeBiochemicalInput.indexOf(term.key) === -1) {
+            query.addKeyValue('biochemical_inputs', term.key);
+            facetObj[term.key].selected = false;
+            facetObj[term.key].link = `?${query.format()}`;
+        } else {
+            query.deleteKeyValue('biochemical_inputs', term.key);
+            facetObj[term.key].selected = true;
+            facetObj[term.key].link = `?${query.format()}`;
+        }
+    });
+    facetObj.All = {};
+    if (initializeBiochemicalInput.length === facet.terms.length) {
+        const query = new QueryString(queryString);
+        query.deleteKeyValue('biochemical_inputs');
+        facetObj.All.selected = true;
+        facetObj.All.link = `?${query.format()}`;
+    } else {
+        const query = new QueryString(queryString);
+        facet.terms.forEach((term) => {
+            if (initializeBiochemicalInput.indexOf(term.key) === -1) {
+                query.addKeyValue('biochemical_inputs', term.key);
+            }
+        });
+        facetObj.All.selected = false;
+        facetObj.All.link = `?${query.format()}`;
+    }
+
+    return (
+        <div className={`facet ${reactContext.location_href.indexOf('encyclopedia') === -1 ? 'hide' : ''}`}>
```

Comment:
Probably easier to use the rarely used `results['@type']` property that you don’t pull in here, just to avoid the highly unusual situation where someone tricks your code by putting a spurious “encyclopedia” in the query string somewhere.

Reply:
I fixed this with the QA update. I think maybe "encyclopedia" could be in the url if an encyclopedia version is chosen from the encyclopedia version facet so this was a good catch.